src/core/config/config-helper.ts

```diff
@@ -48,34 +41,75 @@ export const getDefaultDebugPort = (
   return configuredPort ?? CHROME_DEFAULT_DEBUGGING_PORT;
 };
 
-export const getCustomLauncher = (config: ConfigStore<ProjectConfigSetting>): CustomLauncher => {
-  const configuredLauncher: CustomLauncher = config.get(GeneralConfigSetting.CustomLauncher);
-  const configuredContainerMode: ContainerMode = config.get(GeneralConfigSetting.ContainerMode);
-  const isNonHeadlessMode = !!config.get(GeneralConfigSetting.NonHeadlessModeEnabled);
+export const loadProjectKarmaConfigFile = (projectKarmaConfigPath: string): KarmaConfig => {
+  let loadedConfiguration: any = {};
+  const fakeConfig = {
+    set: (newConfig: any) => {
+      loadedConfiguration = { ...loadedConfiguration, ...newConfig };
+    }
+  };
 
-  const isContainerMode =
-    configuredContainerMode === ContainerMode.Enabled
-      ? true
-      : configuredContainerMode === ContainerMode.Disabled
-      ? false
-      : isDocker();
+  let projectKarmaConfigModule = require(projectKarmaConfigPath); // eslint-disable-line @typescript-eslint/no-var-requires
```

Comment:
This makes sense on the surface, but the extension currently goes to great lengths to not execute the Karma config script except during the actual Karma launch.

This is because being a script and not just a simple config file, the Karma config can (and many actually do) perform arbitrary operations, including things like read (potentially expensive) configuration or other resources, perform local or remote setup or adjustment operations etc, in the assumption that the testing process (Karma) is starting.

This makes it not always completely innocuous or always cheap to execute the Karma config script outside of the necessary instance of the actual Karma process launch. This wouldn't be a problem if it were a JSON file.

Reply:
Do you have any suggestions on how to tackle that in terms of getting the browser type and custom launcher configurations so that we can detect which debugger configuration to use? The actual custom launcher logic can be moved into the karma config loader easily enough, but we need to know about the debugger configuration before that.

Reply:
I don't have any detailed thoughts on how to do that, but I believe (but could be wrong) that since all Chromium browsers use the same Chrome inspector/debugger protocol, the VS Code chrome debugger should work for all (?) chromium browsers as long as they are started with the correct debug port where the debugger is connecting to.

So then, one consideration about this change might be to scope down the level of expanded browser support to just Chromium browsers for now, especially since VS Code only has built-in debugger support for those (Chrome and Edge), with Firefox needing an extra marketplace extension from what I can determine.

On the Karma process side where you've got access to the karma configuration, you could therefore find and select the first (or preferred) Chromium-based browser present in the project's browsers list, if any - or, fallback to current behavior of Chrome if none. For the chromium browser you select, you can set (or update, if already present) the debugger port to the one received from the extension through the environment variable. I believe (but could be wrong) all chromium-based browsers set the debugger port with the same flag as Chrome.

In addition to the current Chrome support, support for chromium-based browsers would expand supported browsers to include Edge, Chromium, and I believe Electron(?), as well as a few others, which would be a meaningful improvement.

Another thought, which might be a better approach - from the extension side, before executing Karma, you could probably glean the full set of browser options available to Karma by looking at the `karma-<browser>-launcher` dependencies in the project `package.json` file. That knowledge might help make some decisions on the extension side (such as browser selection) ahead of launching Karma. In that case, it may also be necessary to validate that the corresponding browser for the chosen launcher is also actually installed. If that turns out to be a viable approach, the debugger configuration could be decided on the extension side as well ahead of the Karma launch, and relayed to it via the current environment variable mechanism.